### PR TITLE
Add property `LiveSstFilesSizeAtTemperature` for tiered storage

### PR DESCRIPTION
**Pull request by jay-zhuang**

### CODE REVIEW

File: db/db_test2.cc [hunk 1]
```diff
@@ -26,6 +26,19 @@ namespace ROCKSDB_NAMESPACE {
...
  public:
   DBTest2() : DBTestBase("db_test2", /*env_do_fsync=*/true) {}
+
+ protected:
+#ifndef ROCKSDB_LITE
+  uint64_t GetSstSizeHelper(Temperature temperature) {
+    std::string prop;
+    bool s = dbfull()->GetProperty(
+        DB::Properties::kLiveSstFilesSizeAtTemperature +
+            std::to_string(static_cast<uint8_t>(temperature)),
```
Comment (siying) [inline db/db_test2.cc:36]: Not a big deal in unit test but we tend to use ToString() defined in string_util.h

File: db/internal_stats.cc [hunk 2]
```diff
@@ -682,6 +689,28 @@ bool InternalStats::HandleBlockCacheEntryStatsMap(
...
+  const auto* vstorage = cfd_->current()->storage_info();
+  for (int level = 0; level < vstorage->num_levels(); level++) {
+    for (const auto& file_meta : vstorage->LevelFiles(level)) {
+      if (static_cast<uint8_t>(file_meta->temperature) == temperature) {
+        size += file_meta->fd.GetFileSize();
+      }
+    }
+  }
+
+  *value = std::to_string(size);
```
Comment (siying) [inline db/internal_stats.cc:710]: Use ToString() instead. Not sure whether it is still necessary, but at least keep it consistent.

File: db/internal_stats.cc [hunk 3]
```diff
@@ -482,6 +486,9 @@ const std::unordered_map<std::string, DBPropertyInfo>
         {DB::Properties::kLiveSstFilesSize,
          {false, nullptr, &InternalStats::HandleLiveSstFilesSize, nullptr,
           nullptr}},
+        {DB::Properties::kLiveSstFilesSizeAtTemperature,
+         {false, &InternalStats::HandleLiveSstFilesSizeAtTemperature, nullptr,
```
Comment (siying) [inline db/internal_stats.cc:490]: Is it possible to turn it to true? Do we need it to be done inside DB mutex? Iterating all files is not so cheap.

**Review by siying — COMMENTED** [review]
Thank you for working on this!

File: db/internal_stats.cc (hunk 3 above)
Comment (jay-zhuang) [reply db/internal_stats.cc]: make sense. Updated.

File: db/internal_stats.cc [hunk 4]
```diff
@@ -482,6 +486,9 @@ const std::unordered_map<std::string, DBPropertyInfo>
         {DB::Properties::kLiveSstFilesSize,
          {false, nullptr, &InternalStats::HandleLiveSstFilesSize, nullptr,
           nullptr}},
+        {DB::Properties::kLiveSstFilesSizeAtTemperature,
+         {true, &InternalStats::HandleLiveSstFilesSizeAtTemperature, nullptr,
```
Comment (siying) [inline db/internal_stats.cc:490]: Is there a reason it is returning string, rather than integer? It feels that it always returns integers.

Comment (jay-zhuang) [reply db/internal_stats.cc:490]: Int one doesn't support suffix, so similar to [`Properties::kNumFilesAtLevelPrefix`](https://github.com/facebook/rocksdb/blob/a63ee24a191b10d9fd615002eb3dea35345aceb0/db/internal_stats.cc#L379) it returns a string.

**Review by siying — APPROVED** [review]

**Review by zhichao-cao — APPROVED** [review]

**Pull request closed without merging**